integration/container/update_linux_test.go

```diff
@@ -4,17 +4,110 @@ import (
 	"bytes"
 	"context"
 	"fmt"
+	"io/ioutil"
 	"strconv"
 	"strings"
 	"testing"
 	"time"
 
 	"github.com/docker/docker/api/types"
 	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/strslice"
 	"github.com/docker/docker/integration/util/request"
 	"github.com/docker/docker/pkg/stdcopy"
+	"github.com/gotestyourself/gotestyourself/poll"
+	"github.com/gotestyourself/gotestyourself/skip"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
+func TestAPIUpdateContainer(t *testing.T) {
+	skip.If(t, testEnv.DaemonInfo.OSType != "linux")
+	skip.If(t, !testEnv.DaemonInfo.MemoryLimit)
+	skip.If(t, !testEnv.DaemonInfo.SwapLimit)
+
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	c, err := client.ContainerCreate(ctx,
+		&container.Config{
+			Cmd:   []string{"top"},
+			Image: "busybox",
+		},
+		&container.HostConfig{
+			Resources: container.Resources{
+				Memory: 200 * 1024 * 1024,
+			},
+		},
+		nil,
+		"",
+	)
+	require.NoError(t, err)
+
+	err = client.ContainerStart(ctx, c.ID, types.ContainerStartOptions{})
+	require.NoError(t, err)
+
+	poll.WaitOn(t, containerIsInState(ctx, client, c.ID, "running"), poll.WithDelay(100*time.Millisecond))
+
+	_, err = client.ContainerUpdate(ctx, c.ID, container.UpdateConfig{
+		Resources: container.Resources{
+			Memory:     314572800,
+			MemorySwap: 524288000,
+		},
+	})
+	require.NoError(t, err)
+
+	inspect, err := client.ContainerInspect(ctx, c.ID)
+	require.NoError(t, err)
+	assert.Equal(t, inspect.HostConfig.Memory, int64(314572800))
+	assert.Equal(t, inspect.HostConfig.MemorySwap, int64(524288000))
+
+	var b bytes.Buffer
```

Comment:
I think everything from this line to the end of the function can be moved out into a helper function, so that it's clear this is part of checking the behaviour.  Maybe something like this:

```go
func getContainerSysFSValue(ctx context.Context, cID string, path string) (string, error) {
}
```

which can be called once for each value